### ISSUE TITLE
Output limit bug

### ISSUE THREAD

**Issue by xavitg** [paul-gauthier/aider#1078]
### Issue

I am getting this problem all the time now when editing a full file:

Model vertex_ai/claude-3-5-sonnet@20240620 has hit a token limit!
Token counts below are approximate.

Input tokens: ~7,123 of 200,000
Output tokens: ~3,017 of 4,096 -- possibly exceeded output limit!
Total tokens: ~10,140 of 200,000

For more info: https://aider.chat/docs/troubleshooting/token-limits.html

As you can see the output tokens are in 3,000 over 4000..

### Version and model info

Aider 0.50.0

**Comment by xavitg** [comment]
This is a problem of the last version 0.50.0 
I changed to an older version: 0.49.0 and it works fine again, but with the last version the output limit is reached before the max limit, as you can see in my 1st message. @paul-gauthier

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

The "~3,017" is an *estimate* of the tokens. It's the LLM which decides if it hit its output limit, not aider.

The problem is that `vertex_ai/claude-3-5-sonnet@20240620` is not listed as supporting "prefill" to allow infinite output. I just submitted a PR to litellm to improve this.

https://github.com/BerriAI/litellm/pull/5203

**Comment by lrtwsp** [comment]
I'm having a similar problem which maybe the same issue. I'm running koboldcpp as my server with: 
`
./koboldcpp-linux-x64-cuda1210 --model ../code/models/Mistral-Large-Instruct-2407-Q4_K_M-00001-of-00002.gguf --port 8080 --usecublas normal --gpulayers 100 --contextsize 16384 --flashattention`

I then initiate aider with the following command: 

`aider --openai-api-key "blarp" --openai-api-base "http://localhost:8080/v1/" --4o`

And while it works for short messages, I eventually end up with this error:

```
Model gpt-4o has hit a token limit!
Token counts below are approximate.

Input tokens: ~4,262 of 128,000
Output tokens: ~20 of 4,096
Total tokens: ~4,282 of 128,000

For more info: https://aider.chat/docs/troubleshooting/token-limits.html
```

The reason I'm using the kobold with the openai server endpoint set to localhost is because I'm behind a corporate zscaler instance that will not connect to litelmm without SSL errors. I get the same error trying to use ollama. I simply can not change this fact. the Openai api + koboldcpp combination works except for the strange token limit issues. I see above you said it's the LLM that decides the token limit. Since I dont have a token generation limit if I just invoke the LLM through a seperate python script, I'm kind of confused as to how to fix this. Suggestions?

**Comment by fry69** [comment]
@lrtwsp Your problem seems to have nothing to do with the original issue. Can you please remove your post from this issue and post it in the separate issue?

@paul-gauthier FYI: Looks like your PR got merged into LiteLLM and is available from version `1.43.13` onward, but it seems current `pipx install aider-chat` installs aider version `0.50.1`, which brings in LiteLLM `1.43.9`

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.

**Comment by lrtwsp** [comment]
> @lrtwsp Your problem seems to have nothing to do with the original issue. Can you please remove your post from this issue and post it in the separate issue? Thank you!
> 
> @paul-gauthier FYI: Looks like your PR got merged into LiteLLM and is available from version `1.43.13` onward, but it seems current `pipx install aider-chat` installs aider version `0.50.1`, which brings in LiteLLM `1.43.9` (without your fix)

Just wanted to say thank you. The upgrade aider-chat seems to have fixed my problems. Thanks again!
